src/main/java/org/lantern/event/PublicIpAndTokenTracker.java

```diff
@@ -10,58 +11,52 @@
 import com.google.inject.Singleton;
 
 /**
- * This class centralizes tracking of the combination of a valid refresh 
- * token and a valid proxy. Several services rely on that combination,
- * including the XMPP server login and connecting to the friends API.
+ * This class centralizes tracking of the combination of a valid refresh token
+ * and a valid proxy. Several services rely on that combination, including the
+ * XMPP server login and connecting to the friends API.
  */
 @Singleton
 public class PublicIpAndTokenTracker {
 
     private final Logger log = LoggerFactory.getLogger(getClass());
-    
-    private boolean hasRefresh = false;
 
-    private boolean hasPublicIp = false;
+    private volatile boolean hasPublicIp = false;
+    private volatile String refresh;
 
```

Comment:
Yeah, I figured I'd leave them volatile in case anyone ever adds code that reads them for any reason.